docs/rfcs/20220408_partition_hibernation.md

```diff
@@ -0,0 +1,277 @@
+- Feature Name: Partition Hibernation
+- Status: draft
+- Start Date: 2022-04-08
+- Authors: John Spray
+- Issue: (one or more # from the issue tracker)
+
+# Executive Summary
+
+Avoid idle partitions using system resources.
+
+## What is being proposed
+
+A new state for partitions, where they exist in the topics table
+but do not have active raft groups.
+
+## Why (short reason)
+
+To enable higher scale when a significant proportion of topics
+are idle.
+
+## How (short plan)
+
+Topics will be automatically placed into the hibernate state when
+they have been idle for more than a configurable threshold.
+
+## Impact
+
+Reduced cost of delivering free/cheap public-facing services, where
+users are likely to create topics for occasional or one-off use.
+
+# Motivation
+
+## Why are we doing this?
+
+Partitions have some runtime resource overhead, even if they aren't being
+used, and haven't been used for some time.
+
+To support public-facing Kafka services on which many users might create
+and then abandon idle topics, we could implement a "hibernate" mode
+for partitions that haven't been used for a long time.
+
+## What use cases does it support?
+
+Implementing a low-cost/no-cost public service, in which developers are likely
```

Comment:
I think this is focusing too much on a niche use case -- really, it supports any use case where users may create a ton of topics / partitions and then end up not using them but also forgetting to clean them up. This could be any dev workflow, ... some public service ..., etc.